### ISSUE TITLE
Vault LDAP Bind Error

### ISSUE THREAD

**Issue by jamesgoodhouse** [concourse/concourse#2679]
## Bug Report

Logs for Concourse are being flooded with the following error:

`Error making API request.\n\nURL: PUT https://<<VAULT_SERVER>>:8200/v1/auth/token/renew-self\nCode: 400. Errors:\n\n* LDAP bind failed: LDAP Result Code 49 \"Invalid Credentials\":`

## Steps to Reproduce

Configure Vault settings for Concourse.

## Expected Results

This error not occur as Vault shouldn't be trying to bind to LDAP.

## Actual Results

This error occurs.

## Additional Context

It is unclear as to the relationship between Vault and LDAP for Concourse. I have read through the docs and failed to find any info as to why this error is happening or why Vault is trying to bind to LDAP.

Also, jobs are able to successfully retrieve secrets from Vault when they run.

## Version Info

* Concourse version: v4.2.1
* Deployment type (BOSH/Docker/binary): Docker
* Infrastructure/IaaS: Custom internal infrastructure
* Browser (if applicable): N/A
* Did this used to work? Unknown

**Comment by xtremerui** [comment]
Could you provide the concourse vault config and the vault server config? If you don't enable ldap on vault or you don't config concourse vault to use LDAP as auth backend by `--vault-auth-backend`, it shouldn't have such error.

**Comment by jamesgoodhouse** [comment]
My Concourse Vault config:

```
CONCOURSE_VAULT_CLIENT_TOKEN=<<MY_TOKEN>>
CONCOURSE_VAULT_INSECURE_SKIP_VERIFY=true
CONCOURSE_VAULT_PATH_PREFIX=/containers/teams/<<OUR_TEAM>>/production
CONCOURSE_VAULT_URL=https://<<OUR_VAULT_SERVER>>:8200
```

I don't know I have access to our Vault server config. What type of settings are you looking for?

I don't have vault-auth-backend set. I tried setting it to `app_role`, but logs still show the same error.

**Comment by xtremerui** [comment]
Given that your jobs are running fine it is probably a config at vault server side. For debugging i would try disable/enable the vault auth for ldap to verify ( but make sure it is safe to do so beforehand).

**Comment by jamesgoodhouse** [comment]
@xtremerui: I assume you mean disable vault auth for ldap on the vault config side of things and not the concourse side of things?

I'm currently unable to do anything on the Concourse side of things to stop it from trying to connect to Vault via LDAP. I checked our vault config and it has LDAP, token, and approle setup. When I configure the vault backend for concourse to either token or approle, it still results in an LDAP error. This is reeking havoc on our LDAP system due to all of failed connection attempts. If I don't set any of the vault stuff up for concourse then i obviously don't have the same issues, but then vault isn't configured to run at all.

**Comment by jmccann** [comment]
I just ran into this myself.  I think the issue lies at https://github.com/concourse/concourse/blob/90a604c940b240a520b284e9a4ba5ad7062c4189/atc/creds/vault/api_client.go#L84

This line of code assumes that all auth methods following `auth/<backend>/login` pattern.  But ldap follows `auth/<backend>/login/:username` pattern as documented at https://www.vaultproject.io/api/auth/ldap/index.html#login-with-ldap-user

I'm going to see how hard it is to build from source and see if I can get this working for myself.  Will post back with any additional findings.

**Comment by pivotal-jamie-klassen** [comment]
I'm almost certain that #3884 fixed this and was released in [v5.3.0](https://github.com/concourse/concourse/releases#v530-note-6).
